test/functional/test_solution.py

```diff
@@ -0,0 +1,439 @@
+        self.ncm.client.associate_health_monitor(proxy.pool['id'], hmconf)
+        # need to associate hm with pool
+        self.proxies.append(proxy)
+        return proxy
+
+    def delete_proxy(self, proxy):
+        # destroy a proxy and all associated objects
+        for healthmonitor in proxy.healthmonitors:
+            self.ncm.client.disassociate_health_monitor(proxy.pool['id'],
+                                                        proxy.healthmonitors[0]['id'])
+            self.ncm.delete_health_monitor(healthmonitor['id'])
+        for member in proxy.members:
+            self.ncm.delete_member(member['id'])
+        self.ncm.delete_vip(proxy.vip['id'])
+        self.wait_for_object_state('vip_id', None,
+                                   self.ncm.show_pool, 'pool', proxy.pool['id'])
+        self.ncm.delete_pool(proxy.pool['id'])
+        # workaround for bug? need to explicitly delete selfip and route domain
+        # for the pool to be deleted, otherwise it is stuck in pending.
+
+    def clear_proxies(self):
+        for vip in self.ncm.list_vips()['vips']:
+            self.ncm.delete_vip(vip['id'])
+        for member in self.ncm.list_members()['members']:
+            self.ncm.delete_member(member['id'])
+        for pool in self.ncm.list_pools()['pools']:
+            for health_monitor in \
+                    self.ncm.list_health_monitors()['health_monitors']:
+                self.ncm.client.disassociate_health_monitor(pool['id'],
+                                                            health_monitor['id'])
+                time.sleep(5)
+                self.ncm.delete_health_monitor(health_monitor['id'])
+            self.wait_for_object_state('vip_id', None,
+                                       self.ncm.show_pool, 'pool', pool['id'])
+            self.ncm.delete_pool(pool['id'])
+        # self.debug()
+
+    def debug(self):
+        print '----- pools -----'
+        pp(self.ncm.list_pools())
+        print '----- vips -----'
+        pp(self.ncm.list_vips())
+        print '----- members -----'
+        pp(self.ncm.list_members())
+        print '----- health monitors -----'
+        pp(self.ncm.list_health_monitors())
+
+
+class LBaaSv2(object):
+    def __init__(self, symbols):
+        self.polling_interval = 1
+        self.symbols = symbols
+        self.ncm = nclientmanager(self.symbols)
+        self.proxies = []
+
+    def clear_loadbalancers(self):
+        for lb in self.ncm.list_loadbalancers()['loadbalancers']:
+            self.ncm.delete_loadbalancer(lb['id'])
+        while self.ncm.list_loadbalancers()['loadbalancers']:
+            time.sleep(1)
+
+    def create_loadbalancer(self):
+        conf = {}
+        for sn in self.ncm.list_subnets()['subnets']:
+            if sn['name'] == self.symbols['client_subnet']:
+                conf = {'vip_subnet_id': sn['id'],
+                          #'lb_method':     'ROUND_ROBIN',
+                          #'protocol':      'HTTP',
```

Comment:
I assume these are commented out because for v2 they are not valid/required to create a LB?